dig.go

```diff
@@ -643,7 +704,7 @@ func (cv connectionVisitor) checkKey(k key, path string) error {
 			"cannot provide %v from %v: already provided by %v",
 			k, path, conflict)
 	}
-	if ps := cv.c.providers[k]; len(ps) > 0 {
+	if ps := cv.c.getRoot().getValueProviders(k.name, k.t); len(ps) > 0 {
```

Comment:
You can likely switch out the Container provided to `connectionVisitor` when
it's instantiated to the root container and leave this as-is.

Reply:
I could, but even then it has to call `getValueProviders` instead of accessing providers directly

Reply:
Oh, yeah, true you'd still have to use getValueProviders to have access to the whole graph.